src/main/java/data/GoodDeed.java

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+public class GoodDeed {
+    private long id;
+    private String title;
+    private String description;
+    private boolean used;
+    private long timestamp;
```

Comment:
Since they aren't changing, can these be marked "final"?

Reply:
I intend to update the used variable depending on what the situation calls for. The goal is to remove a "used" deed from the query to eliminate repeating deeds in a certain time frame. Once the amount of "unused" deeds reaches a given minimum, I would change the values so that the web app could cycle through them again. Would I still need to put final in this case? Also, should id also have final?

Reply:
That makes sense, but since the current change does not show those references it would be better to mark final now and then remove final later when you actually use the variables.

You could also add "setUsed(boolean isUsed)" to this change, and then leave used as not final.  That way it is obvious why that one shouldn't be final.

Basically the goal here  is that each change makes sense by itself.  Though I do get your reasoning.